BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/BeaconUtil.java

```diff
@@ -39,6 +46,126 @@ public static float getAdvertisingRange(int transmissionPower) {
+     * @return Estimated rssi for the given distance
+     */
+    public static int calculateRssiForDistance(Beacon beacon, float distance) {
+        return calculateRssi(distance, beacon.getCalibratedRssi(), beacon.getCalibratedDistance(), BeaconDistanceCalculator.getPathLossParameter());
+    }
+
+    /**
+     * Calculates distances using the reverse <a href="https://en.wikipedia.org/wiki/Log-distance_path_loss_model">log-distance
+     * path loss model</a>.
+     *
+     * @param distance           Distance the beacon should be away
+     * @param calibratedRssi     the RSSI measured at the calibration distance
+     * @param calibratedDistance the distance in meters at which the calibrated RSSI was measured
+     * @param pathLossParameter  the path-loss adjustment parameter
+     * @return Estimated rssi for the given distance
+     */
+    public static int calculateRssi(float distance, float calibratedRssi, int calibratedDistance, float pathLossParameter) {
+        return calculateRssi(distance, BeaconDistanceCalculator.getCalibratedRssiAtOneMeter(calibratedRssi, calibratedDistance), pathLossParameter);
+    }
+
+    /**
+     * Calculates distances using the reverse <a href="https://en.wikipedia.org/wiki/Log-distance_path_loss_model">log-distance
+     * path loss model</a>.
+     *
+     * @param distance          Distance for which a rssi should be estimated
+     * @param calibratedRssi    the RSSI measured at 1m distance
+     * @param pathLossParameter the path-loss adjustment parameter
+     */
+    public static int calculateRssi(float distance, float calibratedRssi, float pathLossParameter) {
+        return (int) ((Math.log(distance) / Math.log(10)) * (10 * pathLossParameter) + calibratedRssi);
+    }
+
+    /**
+     * Please note that if you want a single instance of a beacon instead of adding it to a list you
+     * need to specify the complete type e.g. "IBeacon<IBeaconAdvertisingPacket>".</IBeaconAdvertisingPacket>
+     *
+     * @param beaconClazz            Class token of the specific beacon type
+     * @param advertisingPacketClazz Class token of the specific advertising packet type
+     * @param distance               Distance for which a rssi will be generated
+     * @param <A>                    Specific advertising packet type
+     * @param <B>                    Specific beacon type
+     * @param <CA>                   Class of the specific advertising packet type
+     * @param <CB>                   Class of the specific beacon type
+     * @return Beacon for the specified beacon type with the specified advertising packet type and
+     *         set rssi
+     * @throws ExecutionException If reflections fail
+     */
+    public static <A extends AdvertisingPacket, B extends Beacon<A>, CA extends Class<A>, CB extends Class<B>> B getBeaconsWithAdvertisingPackets(CB beaconClazz, CA advertisingPacketClazz, float distance) throws ExecutionException {
```

Comment:
This should only be in the test package as it's not used in production. Also, rename `Clazz` to `Class`.